### PR TITLE
Improve put and patch when data does not change

### PR DESCRIPTION
**Pull request by thierrydallacroce**

### CODE REVIEW

File: modules/custom/dkan_metastore/tests/src/Unit/ServiceTest.php [hunk 1]
```diff
@@ -112,6 +113,39 @@ public function testPut() {
     $this->assertEquals("1", $info['identifier']);
   }
 
+  /**
+   *
+   */
+  public function testPutModifyIdentifierException() {
```
Comment (fmizzell) [inline modules/custom/dkan_metastore/tests/src/Unit/ServiceTest.php]: Nice!!!

File: modules/custom/dkan_metastore/src/Service.php [hunk 2]
```diff
@@ -286,8 +293,7 @@ public function delete($schema_id, $identifier) {
    */
   private function objectExists($schemaId, $identifier) {
     try {
-      $this->getEngine($schemaId)->get($identifier);
-      return TRUE;
+      return $this->getEngine($schemaId)->get($identifier);
```
Comment (fmizzell) [inline modules/custom/dkan_metastore/src/Service.php]: I am not sure I like changing the semantics of this method to do more than just answer the question "does the object exist?".

File: modules/custom/dkan_metastore/src/Service.php [hunk 3]
```diff
@@ -254,7 +258,10 @@ public function put($schema_id, $identifier, string $data): array {
    */
   public function patch($schema_id, $identifier, $data) {
     $engine = $this->getEngine($schema_id);
-    if ($this->objectExists($schema_id, $identifier)) {
+    if ($existing_data = $this->objectExists($schema_id, $identifier)) {
```
Comment (fmizzell) [inline modules/custom/dkan_metastore/src/Service.php]: This code is exactly the same as the one in the put method. Should we move this "objectHasChanged" logic into its own method?

**Comment by fmizzell** [comment]
@thierrydallacroce Can we look at the complexity issues flagged by codeclimate?

File: modules/custom/dkan_metastore/src/Service.php (hunk 3 above)
Comment (thierrydallacroce) [reply modules/custom/dkan_metastore/src/Service.php]: Thanks, I will follow your suggestion:
- Restore the original `objectExists`
- Create a `objectHasChanged`

**Pull request closed without merging**